lib/pbench/server/database/models/api_keys.py

```diff
@@ -68,14 +71,14 @@ def add(self):
             Database.db_session.commit()
         except Exception as e:
             Database.db_session.rollback()
-            self.logger.error("Can't add {} to DB: {}", str(self), str(e))
-            decode_exc = decode_integrity_error(
-                e, on_duplicate=DuplicateApiKey, on_null=NullKey
+            raise decode_sql_error(
+                e,
+                on_duplicate=DuplicateApiKey,
+                on_null=NullKey,
+                fallback=APIKeyError,
+                operation="add",
+                key=self,
             )
```

Comment:
`operation` and `key` (in the form of the `kwarg` dict) go into the exception text as well as being available as an attribute.

As for the fallback ... yeah, it should, because the base type doesn't have the right constructor. Dang.